target/riscv/op_helper_cheri.c

```diff
@@ -50,25 +50,35 @@
 enum SCRAccessMode {
     SCR_Invalid = 0,
     ASR_Flag = 1,
-    U_Always = (PRV_U + 1) << 1,
-    U_ASR = U_Always | ASR_Flag,
-    S_Always = (PRV_S + 1) << 1,
-    S_ASR = S_Always | ASR_Flag,
-    H_Always = (PRV_H + 1) << 1,
-    H_ASR = H_Always | ASR_Flag,
-    M_Always = (PRV_M + 1) << 1,
-    M_ASR = M_Always | ASR_Flag,
+    ASR_W_Flag = 2,
+    U_Always = (PRV_U + 1) << 2,
+    U_ASR_W = U_Always | ASR_W_Flag,
+    U_ASR = U_ASR_W | ASR_Flag,
```

Comment:
U_ASR is ASR + ASR_W, which is a bit odd (i.e. you're inconsistent between the flag and the aliases what the unsuffixed version means)

Reply:
Also, `U_Always | ASR_Flag` and `U_Always | ASR_Flag | ASR_W_Flag` behave the same, which is a bit confusing too IMO

Reply:
@jrtc27 Would `ASR_RW = ASR_R | ASR_W` be something you are happy with? Then all SCRs other than xTIDC need to have `ASR_RW`.